driver/main.c

```diff
@@ -2161,13 +2211,29 @@ static inline void g_n_tracepoint_hit_inc(void)
 
 TRACEPOINT_PROBE(syscall_enter_probe, struct pt_regs *regs, long id)
 {
-	long table_index;
-	const struct syscall_evt_pair *cur_g_syscall_table = g_syscall_table;
-	bool compat = false;
-#ifdef __NR_socketcall
-	int socketcall_syscall = __NR_socketcall;
-#else
+	struct event_data_t event_data = {};
+	struct syscall_evt_pair event_pair = {};
+	long table_index = 0;
 	int socketcall_syscall = -1;
+
+	/* Just to be extra-safe */
+	if(id < 0)
+	{
+		return;
+	}
+
+	event_data.category = PPMC_SYSCALL;
+	event_data.event_info.syscall_data.regs = regs;
+
+	/* This is used when we preload params */
+	event_data.is_socketcall = false;
+
+	/* By default we are on 64 bit */
+	event_data.event_info.syscall_data.cur_g_syscall_table = g_syscall_table;
+	event_data.compat = false;
```

Comment:
Again, in my dreams, we wouldn't even have the `compat` and `is_socketcall` fields if they're not needed (based on `CONFIG_*`). Out of scope here.

Reply:
A 
man
can
dream.